src/kOS/Screen/TermWindow.cs

```diff
@@ -963,11 +961,20 @@ protected void CheckResizeDrag()
             }
         }
         
-        void ShowCharacterByAscii(char ch, int x, int y, bool reversingScreen, int charWidth, int charHeight, float brightness)
+        void DrawCursorAt(char ch, int x, int y, bool reversingScreen, int charWidth, int charHeight, float brightness)
```

Comment:
You're right.  I saw the method for drawing a character and didn't go back and see that indeed the main display is drawn line by line.   My bad for not doing enough research before commenting.